vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java

```diff
@@ -158,12 +289,250 @@ public void test_updateContentOrder_forNonExistentId() throws BlockDoesNotExistE
+        when(contentBlockRepository.findById(blockId)).thenReturn(Optional.of(contentBlock));
+        Mockito.doThrow(EmptyResultDataAccessException.class).when(choiceBlockRepo).deleteById(blockId);
+        managerToTest.deleteChoiceBlockById(blockId, slideId);
+
+    }
+
+    @Test
+    public void test_updateImageBlock_success1() throws ImageCouldNotBeStoredException {
+        String slideId = "realSlideId";
+        String fileName = "dummyFile";
+        ISlide slide = new Slide();
+        slide.setId(slideId);
+        IVSImage slideContentImage = new VSImage();
+        slideContentImage.setHeight(700);
+        slideContentImage.setWidth(1300);
+        slideContentImage.setFilename(fileName);
+        byte[] image = new byte[1000];
+        IImageBlock imageBlock = new ImageBlock();
+        when(slideManager.getSlide(slideId)).thenReturn(slide);
+        when(imageFactory.createImage(fileName, fileName)).thenReturn(slideContentImage);
+        when(imageRepo.save((VSImage) slideContentImage)).thenReturn((VSImage) slideContentImage);
+        managerToTest.updateImageBlock(imageBlock, image, fileName);
+
+    }
+
+    @Test
+    public void test_updateImageBlock_success2() throws ImageCouldNotBeStoredException {
+
+        IVSImage slideContentImage = new VSImage();
+        slideContentImage.setHeight(700);
+        slideContentImage.setWidth(1300);
+        IImageBlock imageBlock = new ImageBlock();
+        managerToTest.updateImageBlock(imageBlock, slideContentImage);
+
+    }
+
+    @Test
+    public void test_createImageBlock_success() throws ImageCouldNotBeStoredException, FileStorageException {
+        String slideId = "slide1";
+        String fileName = "dummyFile";
+        String contentType = "application/octet-stream";
+        Integer contentOrder = 3;
+        ISlide slide = new Slide();
+        slide.setId(slideId);
+        IVSImage slideContentImage = new VSImage();
+        slideContentImage.setHeight(700);
+        slideContentImage.setWidth(1300);
+        IImageBlock imageBlock = new ImageBlock();
+        imageBlock.setContentOrder(contentOrder);
+        byte[] image = new byte[1000];
+        String relativePathString = "";
+
+        when(slideManager.getSlide(slideId)).thenReturn(slide);
+        when(imageFactory.createImage(fileName, contentType)).thenReturn(slideContentImage);
+        when(storage.storeFile(image, fileName, contentType)).thenReturn(relativePathString);
+        when(imageBlockRepo.save((ImageBlock) imageBlock)).thenReturn((ImageBlock) imageBlock);
+        Mockito.when(imageRepo.save((VSImage) slideContentImage)).thenReturn((VSImage) slideContentImage);
+        Mockito.when(imageBlockFactory.createImageBlock(slide, slideContentImage)).thenReturn(imageBlock);
+        managerToTest.createImageBlock(slideId, image, fileName, contentOrder);
+
+    }
+
+    @Test
+    public void test_createImageBlock2_success() throws ImageCouldNotBeStoredException, FileStorageException {
```

Comment:
method name does not follow naming convention

Reply:
Many methods in that class has got same name, how to address them?

Reply:
the name should express what the outcome of the test is, so not just "success" but "successSomeCondition" expressing what the tested condition is.